### DISCUSSION TITLE
Update HelmRelease on git tag creation

### DISCUSSION THREAD

**Discussion by jpellizzari** [fluxcd/flux2#5121] [Q&A] [answered]
I have a use case where we want our devs to create a tag for their infra changes to deploy to production. For example, they might change some values for their service, which will auto-deploy to pre-production on merge to main. But for prod, we want to the operation of creating a tag to trigger the change. 

Has anyone achieved this or a similar workflow in Flux? We have only been able to do this with imperative CLI calls to create a HelmRelease manually.

**Comment by matheuscscp** [comment, marked as answer]
If you use an `OCIRepository` for your helm charts (https://fluxcd.io/flux/guides/helmreleases/#oci-repository), you can combine it with an `ImageUpdateAutomation`:

https://fluxcd.io/flux/guides/image-update/

Helm docs for OCI repositories: https://helm.sh/docs/topics/registries/

**Reply by jpellizzari** [reply]
Thanks! I will give that a try and report back
